SpaceInvaders/src/Controllers/BulletsController.java

```diff
@@ -0,0 +1,127 @@
+        bullets.remove(d);
+        return true;
+    }
+
+
+    public AlienController CheckBulletsAliensCollision(AlienController alienController){
+        for(int d = 0; d < bullets.size(); d++){
+            if(bullets.get(d).getEnemyBullet() == false){
+                AlliensInBulletPositionCollisionCheck(alienController, d);
+            }
+        }
+        return alienController;
+    }
+
+    private void AlliensInBulletPositionCollisionCheck(AlienController alienController, int d) {
+        Boolean hit = false;
+        for(int c = 0; c < alienController.GetAliens().size(); c++){
+            if(alienController.GetAliens().get(c).getPositionX() + (alienController.GetAliens().get(c).getSize() - 1) >= bullets.get(d).getPositionX() && alienController.GetAliens().get(c).getPositionX() - (alienController.GetAliens().get(c).getSize() - 1) <= bullets.get(d).getPositionX() && alienController.GetAliens().get(c).getPositionY() == bullets.get(d).getPositionY()){
+                ComputeBulletAlienCollision(alienController, d, c);
+                hit = true;
+                break;
+            }
+            if(hit == true) break;
+        }
+        if(hit == false)
+            for(int c = 0; c < alienController.GetAliens().size(); c++){
+                if(alienController.GetAliens().get(c).getPositionX() + (alienController.GetAliens().get(c).getSize() - 1) >= bullets.get(d).getPositionX() && alienController.GetAliens().get(c).getPositionX() - (alienController.GetAliens().get(c).getSize() - 1) <= bullets.get(d).getPositionX() && alienController.GetAliens().get(c).getPositionY() == bullets.get(d).getPositionY() - 1){
+                    ComputeBulletAlienCollision(alienController, d, c);
+                    break;
+                }
+            }
+    }
+
+    private void ComputeBulletAlienCollision(AlienController alienController, int d, int c) {
+        alienController.ChangeShooter(alienController.GetAliens().get(c).getPositionX(), alienController.GetAliens().get(c).getPositionY() - 1);
+        alienController.GetAliens().remove(c);
+        bullets.remove(d);
+    }
+
+    public PlayerController CheckBulletsPlayerCollision(PlayerController playerController){
+        for(int d = 0; d < bullets.size(); d++){
+            if(playerController.getPositionX() == bullets.get(d).getPositionX() && playerController.getPositionY() == bullets.get(d).getPositionY()){
+                playerController.PlayerHit(false);
+                bullets.remove(d);
+                break;
+            }
+            else if(playerController.getPositionX() == bullets.get(d).getPositionX() && playerController.getPositionY() == bullets.get(d).getPositionY() + 1 && bullets.get(d).getEnemyBullet() == true){
+                playerController.PlayerHit(false);
+                bullets.remove(d);
+                break;
+            }
+        }
+        return playerController;
+    }
+
+}

```

Comment:
Bullet controller gali buti, kad turi per daug metodų. gali buti per daug giliai kviečiami metodai Law of demeter